src/python/pants/core/goals/publish.py

```diff
@@ -242,7 +244,10 @@ async def run_publish(console: Console, publish: PublishSubsystem) -> Publish:
             continue
 
         logger.debug(f"Execute {pub.process}")
-        res = await Effect(InteractiveProcessResult, InteractiveProcess, pub.process)
+        res = await Effect(
+            InteractiveProcessResult,
+            {pub.process: InteractiveProcess, local_environment.val: EnvironmentName},
+        )
```

Comment:
It's likely that some `publish` implementations will need to pin the construction of the actual publish command to use back to the local environment... because otherwise they might consume environment variables / etc to construct it "for" a remote environment?

I'm not sure if that suggests that they should each need to figure out their own environment internally instead... but ... maybe?

Reply:
Isn't that what's going on here? The _packaging_ request is using the environment that's needed to successfully construct the package, but the publish command is run locally.